xray/src/inpaint.rs

```diff
@@ -132,36 +132,33 @@ impl<'a> SpatialNodeInpainter<'a> {
         Ok(())
     }
 
-    fn interpolate_inpaint_image_with_right(&self) -> ImageResult<()> {
-        if let (Some((mut current, current_path)), Some((mut right, right_path))) = (
+    fn interpolate_inpaint_image_with(&self, direction: Direction) -> ImageResult<()> {
+        if let (Some((mut current, current_path)), Some((mut neighbor, neighbor_path))) = (
             self.inpaint_image_and_path_from(None)?,
-            self.inpaint_image_and_path_from(Direction::Right)?,
+            self.inpaint_image_and_path_from(direction)?,
         ) {
-            let (width, height) = (current.width() / 2, current.height());
-            interpolate_subimages(
-                &mut right.sub_image(0, 0, width, height),
-                &mut current.sub_image(width, 0, width, height),
-                |i, _| i as f32 / (width - 1) as f32,
-            );
-            current.save(current_path)?;
-            right.save(right_path)?;
-        }
-        Ok(())
-    }
-
-    fn interpolate_inpaint_image_with_bottom(&self) -> ImageResult<()> {
-        if let (Some((mut current, current_path)), Some((mut bottom, bottom_path))) = (
-            self.inpaint_image_and_path_from(None)?,
-            self.inpaint_image_and_path_from(Direction::Bottom)?,
-        ) {
-            let (width, height) = (current.width(), current.height() / 2);
+            let (width, height) = match direction {
+                Direction::Right => (current.width() / 2, current.height()),
+                Direction::Bottom => (current.width(), current.height() / 2),
+                _ => unimplemented!(),
+            };
+            let ((current_x, current_y), (neighbor_x, neighbor_y)) = match direction {
+                Direction::Right => ((width, 0), (0, 0)),
+                Direction::Bottom => ((0, height), (0, 0)),
+                _ => unimplemented!(),
+            };
+            let neighbor_weighting_function: Box<dyn Fn(u32, u32) -> f32> = match direction {
+                Direction::Right => Box::new(|i, _| i as f32 / (width - 1) as f32),
+                Direction::Bottom => Box::new(|_, j| j as f32 / (height - 1) as f32),
+                _ => unimplemented!(),
+            };
             interpolate_subimages(
-                &mut bottom.sub_image(0, 0, width, height),
-                &mut current.sub_image(0, height, width, height),
-                |_, j| j as f32 / (height - 1) as f32,
+                &mut neighbor.sub_image(neighbor_x, neighbor_y, width, height),
```

Comment:
nit: neighbor_x and neighbor_y are crrently always set to 0. Why don't we pass 0's directly here?

Reply:
In case we implemented top/left as well they wouldn't be zero, so I preferred this generic way...

Reply:
makes sense :+1: